# Generated by Django 4.2.11 on 2024-04-18 13:21

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('netbox_inventory', '0006_purchase_status'),
    ]

    operations = [
        migrations.AlterUniqueTogether(
            name='asset',
            unique_together=set(),
        ),
        migrations.AddConstraint(
            model_name='asset',
            constraint=models.UniqueConstraint(fields=('device_type', 'serial'), name='unique_device_type_serial'),
        ),
        migrations.AddConstraint(
            model_name='asset',
            constraint=models.UniqueConstraint(fields=('module_type', 'serial'), name='unique_module_type_serial'),
        ),
        migrations.AddConstraint(
            model_name='asset',
            constraint=models.UniqueConstraint(fields=('inventoryitem_type', 'serial'), name='unique_inventoryitem_type_serial'),
        ),
        migrations.AddConstraint(
            model_name='asset',
            constraint=models.UniqueConstraint(fields=('owner', 'asset_tag'), name='unique_owner_asset_tag'),
        ),
        migrations.AddConstraint(
            model_name='asset',
            constraint=models.UniqueConstraint(models.F('asset_tag'), condition=models.Q(('owner__isnull', True)), name='unique_asset_tag', violation_error_message='Asset with this Asset Tag and no Owner already exists.'),
        ),
    ]
